api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/callback/NoDataCallback.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2014-2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.api.jaxrs.callback;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.hawkular.metrics.core.api.MetricAlreadyExistsException;
+import org.hawkular.metrics.core.impl.mapper.NoResultsException;
+
+import com.google.common.base.Throwables;
+import com.google.common.util.concurrent.FutureCallback;
+
+/**
+ * @author John Sanda
+ */
+public class NoDataCallback<T> implements FutureCallback<T> {
+
+    protected AsyncResponse response;
+
+    public NoDataCallback(AsyncResponse response) {
+        this.response = response;
+    }
+
+    @Override
+    public void onSuccess(Object result) {
+        response.resume(Response.ok().type(MediaType.APPLICATION_JSON_TYPE).build());
+    }
+
+    @Override
+    public void onFailure(Throwable t) {
+        if (t instanceof MetricAlreadyExistsException) {
+            Error errors = new Error("A metric input id already exists " + ":"
+                    + Throwables.getRootCause(t).getMessage());
+            response.resume(Response.status(Status.BAD_REQUEST).entity(errors).type(APPLICATION_JSON_TYPE).build());
```

Comment:
Lets keep this implementation to merge the PR ASAP.  Just want to let you know that this will change for [HWKMETRICS-26 Tracker - REST-API uses return codes incorrectly](https://issues.jboss.org/browse/HWKMETRICS-26).